plugin-maven/src/main/java/com/diffplug/spotless/maven/kotlin/Ktlint.java

```diff
@@ -43,6 +44,12 @@ public class Ktlint implements FormatterStepFactory {
 	public FormatterStep newFormatterStep(final FormatterStepConfig stepConfig) {
 		String ktlintVersion = version != null ? version : KtLintStep.defaultVersion();
 		FileSignature configPath = null;
+		if (editorConfigPath == null) {
+			File defaultEditorConfig = new File(".editorconfig");
+			if (defaultEditorConfig.exists()) {
+				editorConfigPath = defaultEditorConfig.getPath();
+			}
+		}
```

Comment:
Will load `EMPTY_EDITOR_CONFIG_DEFAULTS` before

https://github.com/diffplug/spotless/blob/cf6c76f83003c290dbbaaadf4d83dba14550e158/lib/src/compatKtLint1Dot0Dot0/java/com/diffplug/spotless/glue/ktlint/compat/KtLintCompat1Dot0Dot0Adapter.java#L96-L101

But this should parse the root .editorconfig, I forget the exact position, let me check.

Reply:
You are right! `KtLintRuleEngine` will load the `.editorconfig`.

However I believe we still need to load it ourselves as well in order to check if it sets `ktlint_code_style` in `KtLintCompat1Dot0Dot0Adapter`. Otherwise we will run into the original issue: `ktlint_code_style` will be overridden to its default value (`intellij_idea`) when the `editorConfigOverride` map is non-empty but does not define it. This is also what `plugin-gradle` does: https://github.com/diffplug/spotless/blob/add4c1e1bb45a01e71e7075ab0fdcb5e91d4fa65/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/BaseKotlinExtension.java#L159-L160